authenticate.go

```diff
@@ -200,6 +201,51 @@ func (c *OktaClient) handleMFAChallenge(transaction api.AuthenticationTransactio
 		}
 		return c.handleAuthUserFlow(newTransaction, false)
 
+	case factors.FactorTypePush:
+		var newTransaction api.AuthenticationTransaction
+		// Sends a request to Okta to push a notification to user's device
+		verifyReq := api.FactorVerifyPush{
+			FactorVerify: api.FactorVerify{
+				StateToken: transaction.StateToken,
+			},
+		}
+		transaction, apiError, err := c.sendTransactionRequest(transaction.Links.Next.HREF, &verifyReq)
+		if err != nil {
+			return c.cancelCurrentFactorWithErrorMessage(transaction, "Cancelled")
+		}
+		if apiError != nil {
+			return c.cancelCurrentFactorWithErrorMessage(transaction, apiError.ErrorSummary)
+		}
+
+		// Prompt user to check their device for an Okta Verify notification
+		c.prompts.VerifyPush()
+
+		// Setup and begin constant backoff policy that retries every 3 seconds with a maximum of 10 attempts (timeout after 30 seconds)
+		backoffPolicy := backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 10)
```

Comment:
@srypher @JackWink Just for my understanding, you guys are doing this backoff manually because the Push factor response doesn't return a `timeoutSeconds` field like U2F and the Okta implementation of Push just keeps you in the `WAITING` state and never expires? 

If that is the case, what do you think about being able to allow the end user to configure the max deadline (and if not set, default to 30 seconds)? Might be out of scope for this PR, but I figure this would be a nice thing to expose as a library.

Additionally, would throw some docstrings in here explaining that as well.

Reply:
Yeah -- I'd look at this PR as still a WIP, we're trying to figure out the edge cases and handling still. That's a good suggestion though about making the timeout configurable.